### DISCUSSION TITLE
A way to search local lists and online lists for subdomains.

### DISCUSSION THREAD

**Discussion by ZeroDot1** [projectdiscovery/subfinder#472] [General]
**Is your feature request related to a problem? Please describe.**
It is sometimes frustrating when you have many lists of helpful subdomains that you can't include directly in subfinder to use as a source when searching for subdomains.

**Describe the solution you'd like**
We should have a possibility to search local lists and online lists for subdomains.
It should be possible to store the paths to the files with the subdomains in the configuration file so that you can quickly search for subdomains without much effort.

Configuration example:
```
#-- externalsources
- url "https://zerodot1.gitlab.io/CoinBlockerLists/hosts"
- file "/database/coinblockerlists/hosts"
- url "ftp.database.com/files/list.txt"
```

This makes it possible to use already existing data more efficiently.
And it also allows searching for subdomains offline (e.g. if you don't have internet).

A search for subdomains could then look like this: 
`/go/bin/subfinder' -config '/Config/subfinder/config.yaml' -externalsources -d coinhive.com`

With the parameter `--externalsources` it should be possible to search the sources stored in the configuration file in addition to the other sources.
It should be possible to search files in the hosts format and in the txt format.

**Comment by ehsandeep** [comment]
This is more aligned with what pointed out here https://github.com/projectdiscovery/subfinder/issues/309 with addition to offline support, basically, we need to add support in subfinder where users can self define and add sources easily.

Thank you @ZeroDot1 we will discuss this and see if/when we can add support for this.

**Comment by ZeroDot1** [comment]
@bauthard Thank you very much, and thank you for your good work I appreciate it very much. :+1:
